tests/layer_tests/tensorflow_tests/test_tf_MatrixInverse.py

```diff
@@ -0,0 +1,47 @@
+# Copyright (C) 2018-2024 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+import numpy as np
+import pytest
+import tensorflow as tf
+from common.tf_layer_test_class import CommonTFLayerTest
+
+class TestMatrixInverse(CommonTFLayerTest):
+    def _prepare_input(self, inputs_info):
+        assert 'input:0' in inputs_info
+        input_shape = inputs_info['input:0']
+        inputs_data = {}
+        inputs_data['input:0'] = self._generate_invertible_matrix(input_shape)
+        return inputs_data
+    
+    def _generate_invertible_matrix(self, shape):
+        while True:
+            matrix = np.random.rand(*shape).astype(np.float32)
+             # if the matrix has correct rank, and the determinat is not zero, then it is invertible
+            if np.linalg.matrix_rank(matrix) == shape[0] and np.linalg.det(matrix) != 0:
+                return matrix
```

Comment:
please have a pool of prepared invertible matrices of different shapes and use choise from numpy to select them randomly.
Here I am not sure how long to wait generation of invertible matrices.

Reply:
Your idea makes sense, I developed a pool of prepared invertible matrices.